### PR TITLE
`Discussion`: Improve design of posting headers and footers

### PR DESCRIPTION
**Pull request by canberkanar**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context
This PR aims to enhance the design of the Post and AnswerPost components since suggestions were made about the current style.

### Description

- To make the design more consistent, we move the emoji picker for Posts to the left-hand side of the Pin and Archive buttons. This way, the picker is positioned right after the reactions users attach to Posts, the way it is already done for AnswerPosts. (instructor account needed to observe the Pin & Archive buttons)
- We integrate the "Answer Now" button to the emoji bar of the last AnswerPost of a Post if it was already answered. This way we reduce the height the Post and its' AnswerPosts take, reducing the scrolling users perform to reach information.
- We redesign the headers of Posts and AnswerPosts to stress the author more and make it easier to read together with the Date/Time a Posting was created.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Lecture with Discussions enabled

1. Log in to Artemis as Instructor
2. Navigate to Course Discussions
3. Verify that the emoji picker is on the left-hand side of the pin and archive buttons.
4. Verify that the "Reply Now" button is integrated to the emoji-bar of the last Answer of a Post.
5. Verify that the Post and AnswerPost's author name is more clearly visible than the previous versions.
6. Select an answer to a post as "resolves post" by checking the checkmark at the top-right corner.
7. Verify that Answers are reordered accordingly. (Answers that resolve should appear at the very top, the rest should be ordered according to creation date)
8. Log in as a student.
9. React to a Post Answer with multiple answers.
10. Make sure nothing unexpected happens.
11. While keeping the instructor tab open, send/edit an answer as a student and verify that the WebSocket subscription still works correctly and AnswerPosts are ordered once again by observing the page from the instructor's tab. (no refreshing of the page is needed to see incoming/updated answers.)

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage
Same as before

### Screenshots

image
image

### CODE REVIEW

**Review by nadinesamyy — DISMISSED** [review]
tested on ts3, everything works as described and expected

**Review by alaraozdenler — APPROVED** [review]
Tested on TS1, works as described

**Review by SimonPaul02 — DISMISSED** [review]
Tested on TS1, everything works!!
But I found a different issue: You can edit and delete comments from other people, even as a student

**Review by yuling-sun — DISMISSED** [review]
Tested on TS1. Changes worked as described.

**Review by babyygemperor — APPROVED** [review]
Tested on TS1, works exactly like described

**Comment by canberkanar** [comment]
> Tested on TS1, everything works!! But I found a different issue: You can edit and delete comments from other people, even as a student

Thank you for your feedback. The issue is probably related to a configuration problem on the test server, as mentioned here.
https://ls1tum.slack.com/archives/C012NFRM76F/p1653160446025809

**Review by max-bergmann — DISMISSED** [review]
Tested on TS1. Works as expected

**Review by ole-ve — APPROVED** [review]
Tested in the testing session on TS1, works as described.

**Review by Lugitan — DISMISSED** [review]
Tested in the testing session on TS1, works as described.

**Review by akesfeden — APPROVED** [review]
Tested on TS1 during testing session as an Instructor, works as expected and looks good 👍

**Review by manuelmanso — APPROVED** [review]
Tested on TS1 during testing session, looks good!

**Review by minrows — DISMISSED** [review]
Tested in the testing session on TS1, works as described.

File: src/main/webapp/app/shared/metis/posting-reactions-bar/post-reactions-bar/post-reactions-bar.component.html [hunk 1]
```diff
@@ -65,6 +43,28 @@
             [darkMode]="darkModeEmojiPicker"
         ></emoji-mart>
     </ng-template>
+    <div class="col-auto p-0" *ngIf="displayPriority === DisplayPriority.PINNED || currentUserIsAtLeastTutor">
+        <button
+            class="reaction-button pin"
+            [class.reaction-button--not-hoverable]="!currentUserIsAtLeastTutor"
+            [class.reaction-button--reacted]="displayPriority === DisplayPriority.PINNED && currentUserIsAtLeastTutor"
+            [disabled]="!currentUserIsAtLeastTutor"
+            (click)="currentUserIsAtLeastTutor && togglePin()"
```
Comment (bassner) [inline src/main/webapp/app/shared/metis/posting-reactions-bar/post-reactions-bar/post-reactions-bar.component.html:52]: Is this required? 🤔 
Seems like the button is disabled already, is there a reason for this check?
```suggestion
            (click)="togglePin()"
```

**Review by bassner — CHANGES_REQUESTED** [review]
Reviewed the code, looks good! Just have two minor remarks

File: src/main/webapp/app/shared/metis/posting-reactions-bar/answer-post-reactions-bar/answer-post-reactions-bar.component.html [hunk 2]
```diff
@@ -43,4 +43,11 @@
             [darkMode]="darkModeEmojiPicker"
         ></emoji-mart>
     </ng-template>
+    <!-- add new answer to expanded discussion -->
+    <div class="col-auto p-0" *ngIf="isLastAnswer">
+        <button id="answerNowButton" class="reaction-button clickable" (click)="openPostingCreateEditModal.emit()">
```
[the remark this answers is not in the document]
Comment (canberkanar) [reply src/main/webapp/app/shared/metis/posting-reactions-bar/answer-post-reactions-bar/answer-post-reactions-bar.component.html:48]: Yes, it is possible, but they would be in different components. Is it a convention error here to have the same ID for these elements, can it cause severe problems?

[the remark this answers is not in the document]
Comment (bassner) [reply src/main/webapp/app/shared/metis/posting-reactions-bar/answer-post-reactions-bar/answer-post-reactions-bar.component.html:48]: It doesn't matter if it's in different components - the browser and the DOM are not aware of components, they just see a big tree of HTML. I think in practice it's tolerated by the browser (and prints a warning iirc), but for example any calls to ``document.getElementById()`` would only return the first element. Nevertheless, it's still a bad practice that should be avoided

**Comment by canberkanar** [comment]
There was a minor change for testing convention. For this I don't think that another manual testing is necessary.

**Review by bassner — APPROVED** [review]
Thank you for the fixes. LGTM! 👍 
Can approve that manual retesting is not necessary. The PR had like 8 tester confirmations before the last small change.

**Pull request merged**